include/aspect/particle/generator/interface.h

```diff
@@ -43,6 +43,7 @@ namespace aspect
       /**
        * Exception denoting a division by zero.
        */
+#if DEAL_II_VERSION_GTE(8,3,0)
       DeclExceptionMsg (ExcParticlePointNotInDomain,
                         "You requested to generate a particle at a position that "
```

Comment:
Clever workaround. Alternatively, you could just use `DeclException1` and make the position of the particle something you output.

Thinking about it, this would require knowing the space dimension of the `Point<dim>` argument. That seems unnecessarily complicated. Forget about it.